lib/src/barback/dartdevc/dartdevc_resource_transformer.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:barback/barback.dart';
+import 'package:cli_util/cli_util.dart' as cli_util;
+import 'package:path/path.dart' as p;
+
+import '../../io.dart';
+
+/// Copies the `dart_sdk.js` and `require.js` amd files from the SDK into each
+/// entrypoint dir.
+class DartDevcResourceTransformer extends AggregateTransformer
+    implements DeclaringAggregateTransformer {
+  /// Group files by output directory, skipping `lib` and `bin`.
+  @override
+  String classifyPrimary(AssetId id) {
+    if (p.extension(id.path) != '.dart') return null;
+    var dir = topLevelDir(id.path);
+    if (dir == 'lib' || dir == 'bin') return null;
+    return p.url.dirname(id.path);
+  }
+
+  @override
+  Future apply(AggregateTransform transform) async {
+    var sdk = cli_util.getSdkDir();
```

Comment:
It makes me sad that this isn't a getter named `sdkDir`, but I guess that's out of scope for this patch.